paymentsheet/src/main/java/com/stripe/android/common/ui/BottomSheet.kt

```diff
@@ -0,0 +1,128 @@
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ModalBottomSheetDefaults
+import androidx.compose.material.ModalBottomSheetLayout
+import androidx.compose.material.ModalBottomSheetState
+import androidx.compose.material.ModalBottomSheetValue
+import androidx.compose.material.ModalBottomSheetValue.Expanded
+import androidx.compose.material.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import com.google.accompanist.systemuicontroller.rememberSystemUiController
+import kotlinx.coroutines.flow.first
+
+@OptIn(ExperimentalMaterialApi::class)
+internal class BottomSheetState(
+    val modalBottomSheetState: ModalBottomSheetState,
+) {
+
+    suspend fun show() {
+        modalBottomSheetState.show()
+    }
+
+    suspend fun awaitDismissal() {
+        snapshotFlow { modalBottomSheetState.isVisible }.first { isVisible -> !isVisible }
+    }
+
+    suspend fun hide() {
+        modalBottomSheetState.hide()
+    }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun rememberBottomSheetState(
+    confirmValueChange: (ModalBottomSheetValue) -> Boolean = { true },
+): BottomSheetState {
+    val modalBottomSheetState = rememberModalBottomSheetState(
+        initialValue = ModalBottomSheetValue.Hidden,
+        confirmValueChange = confirmValueChange,
+        skipHalfExpanded = true,
+        animationSpec = tween(),
+    )
+
+    return remember {
+        BottomSheetState(modalBottomSheetState)
+    }
+}
+
+/**
+ * Renders the provided [sheetContent] in a modal bottom sheet.
+ *
+ * @param state The [BottomSheetState] that controls the visibility of the bottom sheet.
+ * @param onShow Called when the bottom sheet is displayed for the first time. This might be used to
+ * navigate to a specific screen.
+ * @param onDismissed Called when the user dismisses the bottom sheet by swiping down. You should
+ * inform your view model about this change.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun BottomSheet(
+    state: BottomSheetState,
+    modifier: Modifier = Modifier,
+    onDismissed: () -> Unit,
+    onShow: () -> Unit = {},
+    sheetContent: @Composable ColumnScope.() -> Unit,
+) {
+    val systemUiController = rememberSystemUiController()
+    val scrimColor = ModalBottomSheetDefaults.scrimColor
+
+    val isExpanded = state.modalBottomSheetState.targetValue == Expanded
+
+    val statusBarColorAlpha by animateFloatAsState(
+        targetValue = if (isExpanded) scrimColor.alpha else 0f,
+        animationSpec = tween(),
+        label = "StatusBarColorAlpha",
+    )
```

Comment:
Found this necessary to make the status bar color change in sync with the bottom sheet background scrim.